include/openthread-types.h

```diff
@@ -395,6 +395,25 @@ typedef enum
 } otDeviceRole;
 
 /**
+ * This structure holds diagnostic information for a Thread Child
+ *
+ */
+typedef struct
+{
+    otExtAddress   mExtAddress;            ///< IEEE 802.15.4 Extended Address
+    uint32_t       mTimeout;               ///< Timeout
+    uint16_t       mRloc16;                ///< RLOC16
+    uint8_t        mChildId;               ///< Router ID
+    uint8_t        mLinkQualityIn;         ///< Link Quality In
+    uint8_t        mNetworkDataVersion;    ///< Network Data Version
+    uint8_t        mAge;                   ///< Time last heard
+    bool           mRxOnWhenIdle : 1;      ///< rx-on-when-idle
+    bool           mSecureDataRequest : 1; ///< Secure Data Requests
+    bool           mFullFunction : 1;      ///< Full Function Device
+    bool           mFullNetworkData : 1;   ///< Full Network Data
+} otChildInfo;
+
```

Comment:
@jwhui, does it make sense to also add the average rssi to the `otChildInfo` struct?